components/automate-ui/src/app/components/create-object-modal/create-object-modal.component.ts

```diff
@@ -67,12 +80,15 @@ export class CreateObjectModalComponent implements OnInit, OnDestroy, OnChanges
   ngOnChanges(changes: SimpleChanges): void {
     if (changes.visible && (changes.visible.currentValue as boolean)) {
 
-      this.store.dispatch(new GetPolicies()); // refresh in case of updates
+      if (this.objectNoun === 'token') {
+        this.store.dispatch(new GetPolicies()); // refresh in case of updates
+      }
 
-      Object.values(this.projects).forEach(p => p.checked = false); // reset projects
+      this.checkedProjectIDs = []; // reset projects
       this.projectsUpdatedEvent.emit();
 
-      Object.values(this.policies).forEach(p => p.checked = false); // reset policies
+      this.policies.forEach(policy =>
+        policy.itemList.forEach(p => p.checked = false)); // reset policies
```

Comment:
Projects and policies are different because projects are fed to ProjectsDropdownComponent that has some further handling before passing on to a child ResourceDropdownComponent. Policies, on the other hand, are fed directly to ResourceDropdownComponent so all processing is here.